<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Transrate by HibberdLab</title>
    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
    <script src="javascripts/main.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

  </head>
  <body>

      <header>
        <h1>Transrate</h1>
        <p>Quality assessment and comparison of transcriptome assemblies</p>
      </header>

      <div id="banner">
        <span id="logo"></span>

        <a href="https://github.com/HibberdLab/transrate" class="button fork"><strong>View On GitHub</strong></a>
        <div class="downloads">
          <span>Downloads:</span>
          <ul>
            <li><a href="https://github.com/HibberdLab/transrate/zipball/master" class="button">ZIP</a></li>
            <li><a href="https://github.com/HibberdLab/transrate/tarball/master" class="button">TAR</a></li>
          </ul>
        </div>
      </div><!-- end banner -->

    <div class="wrapper">
      <nav>
        <ul></ul>
      </nav>
      <section>
        <h3>
<a name="transrate-helps-you-understand-your-transcriptome-assembly" class="anchor" href="#transrate-helps-you-understand-your-transcriptome-assembly"><span class="octicon octicon-link"></span></a>Transrate helps you understand your transcriptome assembly</h3>

<p>Transcriptome assembly is hard. The algorithms are complex, the data is messy, and it's often not clear how to determine whether an assembly is suitable for answering a biological question. Transrate tries to help by examining your assembly in detail and providing insight about its quality and usefulness in various situations. This can allow you to choose between assemblers and parameters, and helps you decide when to stop trying to improve the assembly.</p>

<h3>
<a name="overview" class="anchor" href="#overview"><span class="octicon octicon-link"></span></a>Overview</h3>

<p>Transrate analyses an assembly in several key ways:</p>

<ul>
<li>by inspecting the contigs themselves</li>
<li>by mapping reads to the contigs and inspecting the alignments</li>
<li>by aligning the contigs against proteins from a related species and inspecting the alignments</li>
</ul><p>The individual metrics will eventually be explained here.</p>

<h3>
<a name="the-command-line-interface" class="anchor" href="#the-command-line-interface"><span class="octicon octicon-link"></span></a>The command-line interface</h3>

<div class="highlight highlight-bash"><pre>  Transrate v0.1.0 by Richard Smith &lt;rds45@cam.ac.uk&gt;

  DESCRIPTION:
  Analyse a de-novo transcriptome
  assembly using three kinds of metrics:

  1. contig-based
  2. <span class="nb">read</span>-mapping <span class="o">(</span><span class="k">if</span> --left and --right are provided<span class="o">)</span>
  3. reference-based <span class="o">(</span><span class="k">if</span> --reference is provided<span class="o">)</span>

  Bug reports and feature requests at:
  http://github.com/blahah/transrate

  USAGE:
  transrate &lt;options&gt;

  EXAMPLES:
  transrate --assembly contigs.fa --reference Athaliana_protein.fa --threads 8

  OPTIONS:
    --assembly, -a &lt;s&gt;:   assembly file<span class="o">(</span>s<span class="o">)</span> in FASTA format, comma-separated
   --reference, -r &lt;s&gt;:   reference proteome file in FASTA format
        --left, -l &lt;s&gt;:   left reads file in FASTQ format
       --right, -i &lt;s&gt;:   right reads file in FASTQ format
  --insertsize, -n &lt;i&gt;:   mean insert size <span class="o">(</span>default: 200<span class="o">)</span>
    --insertsd, -s &lt;i&gt;:   insert size standard deviation <span class="o">(</span>default: 50<span class="o">)</span>
     --threads, -t &lt;i&gt;:   number of threads to use <span class="o">(</span>default: 8<span class="o">)</span>
     --outfile, -o &lt;s&gt;:   filename to use <span class="k">for</span> CSV output <span class="o">(</span>default: transate_results.csv<span class="o">)</span>
           --quiet, -q:   don<span class="err">'</span>t print results - only save to file
         --profile, -p:   debug option: profile the code as it runs
         --version, -v:   Print version and <span class="nb">exit</span>
            --help, -h:   Show this message
</pre></div>

<h3>
<a name="authors-and-contributors" class="anchor" href="#authors-and-contributors"><span class="octicon octicon-link"></span></a>Authors and Contributors</h3>

<p>Transrate was conceived and led by Richard Smith-Unna (RSU) and written by RSU and Chris Boursnell under the guidance of Professor Julian Hibberd at the University of Cambridge. Steve Kelly (Oxford) and Matt MacManes (University of New Hampshire) helped conceive and refine metrics.</p>

<p>We welcome collaboration - if you'd like help with your transcriptome project please send an email: <a href="mailto:rds45@cam.ac.uk">rds45@cam.ac.uk</a>. We also welcome contributions to the code base - please see the <a href="https://github.com/Blahah/transrate">README in the Github repo</a>.</p>

<h3>
<a name="support-or-contact" class="anchor" href="#support-or-contact"><span class="octicon octicon-link"></span></a>Support or Contact</h3>

<p>If you're having trouble with transrate, please first consult the documentation. If you can't find the answer to your question, you can search the <a href="https://groups.google.com/forum/#!forum/transrate-users">user forum</a>. If you still can't find the answer, please post details of your question to the user forum. If you're sure the problem you're encountering is a bug, please post it to the <a href="https://github.com/Blahah/transrate/issues?state=open">issue tracker</a> and we'll fix it.</p>
      </section>
      <footer>
        <p>Project maintained by <a href="https://github.com/HibberdLab">HibberdLab</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://twitter.com/michigangraham">mattgraham</a></small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
              <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-52559492-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>