
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Transrate</title>
    <link rel="stylesheet" href="/transrate/assets/themes/leap-day/stylesheets/styles.css">
    <link rel="stylesheet" href="/transrate/assets/themes/leap-day/stylesheets/pygment_trac.css">
    <link href="/transrate/favico.png" rel="shortcut icon" type="image/x-icon">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
    <script src="/transrate/assets/themes/leap-day/javascripts/main.js"></script>
    <script src="/transrate/assets/themes/leap-day/javascripts/ggl-analytics.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

  </head>
  <body>

      <header>
        <img class="headlogo" src="/transrate/assets/themes/leap-day/images/mainlogo.png">
      </header>

      <div id="banner">
        <div style="width: 650px; margin-left: auto; margin-right: auto;">
          <a href="index.html" class="button"><strong>Home</strong></a>
          <a href="installation.html" class="button"><strong>Installation</strong></a>
          <a href="getting_started.html" class="button"><strong>Getting Started</strong></a>
          <a href="metrics.html" class="button"><strong>Metrics</strong></a>
          <a href="https://github.com/Blahah/transrate" class="button logo" style="float: right;"><strong>View on Github</strong></a>
        </div>

      </div><!-- end banner -->

    <div class="wrapper">
      <nav>
        <ul></ul>
      </nav>
      <section>

<div class="page-header">
  <h1>Transrate <small></small></h1>
</div>

<div class="row">
  <div class="span14">
    <p>Transcriptome assembly is hard. The algorithms are complex, the data are messy, and it&#39;s often not clear how to determine whether an assembly is suitable for answering a biological question.</p>

<p>Transrate helps by examining your assembly in detail and providing insight about its quality and usefulness in various situations. This can allow you to choose between assemblers and parameters, and helps you decide when to stop trying to improve the assembly.</p>

<blockquote>
<p>NOTE: this is the documentation for transrate v1.0.0 beta1</p>
</blockquote>

<h2 id="toc_0">Overview</h2>

<p>Transrate analyses a transcriptome assembly in three key ways:</p>

<ol>
<li>by inspecting the contigs themselves</li>
<li>by mapping reads to the contigs and inspecting the alignments</li>
<li>by aligning the contigs against proteins from a related species and inspecting the alignments</li>
</ol>

<p><strong>By far</strong> the most useful metrics are those based on read mapping, in particular you should pay attention to the <em>Transrate score</em> and the individual <em>contig scores</em>.</p>

<p>You can read about these in detail on the <a href="metrics.html">transcriptome assembly metrics</a> page.</p>

<h2 id="toc_1">Installation</h2>

<p>See <a href="installation.html">the installation guide</a>.</p>

<h2 id="toc_2">The command-line interface</h2>

<p>Running <code>transrate --help</code> will show you the command-line interface:</p>
<div class="highlight"><pre><code class="text language-text" data-lang="text">  Transrate v1.0.0.beta1
  by Richard Smith-Unna &lt;rds45@cam.ac.uk&gt; and Chris Boursnell

  DESCRIPTION:
  Analyse a de-novo transcriptome assembly using three kinds of metrics:

  1. contig-based
  2. read-mapping (if --left and --right are provided)
  3. reference-based (if --reference is provided)

  Bug reports and feature requests at:
  http://github.com/blahah/transrate

  USAGE:
  transrate &lt;options&gt;

  EXAMPLES:
  # check dependencies and install any that are missing
  transrate --install-deps
  # get the transrate score for the assembly and each contig
  transrate --assembly contigs.fa --left left.fq --right right.fq
  # contig metrics only
  transrate --assembly contigs.fa
  # contig and reference-based metrics with 8 threads
  transrate --assembly contigs.fa --reference Athaliana_protein.fa --threads 8
  # contig and read-based metrics for two assemblies with 32 threads
  transrate --assembly one.fa,two.fa --left l.fq --right r.fq --threads 32

  OPTIONS:
   --assembly, -a &lt;s&gt;:   assembly file(s) in FASTA format, comma-separated
  --reference, -r &lt;s&gt;:   reference proteome file in FASTA format
       --left, -l &lt;s&gt;:   left reads file in FASTQ format
      --right, -i &lt;s&gt;:   right reads file in FASTQ format
    --threads, -t &lt;i&gt;:   number of threads to use (default: 8)
    --outfile, -o &lt;s&gt;:   prefix filename to use for CSV output (default:
                         transrate)
   --loglevel, -g &lt;s&gt;:   the amount of information to print. one of [error,
                         info, warn, debug] (default: info)
   --install-deps, -n:   install any missing dependencies
        --version, -v:   Print version and exit
           --help, -h:   Show this message
</code></pre></div>
<p>See the <a href="getting_started.html">getting started guide</a> for more instructions.</p>

<h2 id="toc_3">Authors and Contributors</h2>

<p>Transrate was conceived and led by Richard Smith-Unna (RSU) and written by RSU and Chris Boursnell under the guidance of  <a href="http://www.stevekellylab.com/">Steve Kelly</a> (Oxford) and <a href="http://hibberdlab.com">Professor Julian Hibberd</a> (University of Cambridge). <a href="http://genomebio.org/">Matt MacManes</a> (University of New Hampshire) helped conceive and refine metrics.</p>

<p>We welcome collaboration - if you&#39;d like help with your transcriptome project please send an email to <a href="mailto:rds45@cam.ac.uk">rds45@cam.ac.uk</a>.</p>

<p>We also welcome contributions to the code base - please see the <a href="https://github.com/Blahah/transrate">README in the Github repo</a>.</p>

<h2 id="toc_4">Get Support or Contact</h2>

<p>If you&#39;re having trouble with transrate, please first consult the documentation.</p>

<p>If you can&#39;t find the answer to your question, you can search the <a href="https://groups.google.com/forum/#!forum/transrate-users">user forum</a>. If you still can&#39;t find the answer, please make a new post to the user forum describing the trouble you&#39;re having.</p>

<p>For urgent help, or just to chat to us, please come to the <a href="https://gitter.im/Blahah/transrate">Transrate chat room</a>.</p>

<p>If you&#39;re sure the problem you&#39;re encountering is a bug, please post it to the <a href="https://github.com/Blahah/transrate/issues?state=open">issue tracker</a> and we&#39;ll fix it.</p>

<h2 id="toc_5">Citation, License &amp; Copyright</h2>

<p>Transrate is research software. We therefore require that users cite Transrate if they use it in a publication. Please use the citation:</p>

<blockquote>
<p>Transrate v1.0.0alpha (2014) RD Smith-Unna, C Boursnell, S Kelly and JM Hibberd.      <a href="http://hibberdlab.com/transrate">http://hibberdlab.com/transrate</a>. DOI: <a href="http://dx.doi.org/10.5281/zenodo.10799">10.5281/zenodo.10799</a>.</p>
</blockquote>

<p>Transrate is free open source software, released under the <a href="http://blahah.mit-license.org/">MIT license</a>.</p>

<p>© Richard Smith-Unna 2014.</p>

  </div>
</div>


      </section>
      <footer>
        <p style="margin-bottom: 5px;">Install Transrate:</p><small><pre style="margin-top=-15px;">gem install transrate</pre></small>
        <p>Current version:<br><a href="https://rubygems.org/gems/transrate"><img src="http://img.shields.io/gem/v/transrate.svg"></a></p>
        <p>Test coverage:<br><a href="https://coveralls.io/r/Blahah/transrate"><img src="http://img.shields.io/coveralls/Blahah/transrate.svg"></a></p>
        <p>Project maintained by<br><a href="mailto:rds45@cam.ac.uk">Richard Smith-Unna</a></p>
        <p><small>Developed with ♥ and Ruby<br>in the <a href="http://hibberdlab.com">Hibberd Lab</a></small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><!--<![endif]-->
  </body>
</html>
